src/components/Game/Game.tsx

```diff
@@ -67,50 +65,44 @@ export const Game = () => {
   const onFirstSubmit = async () => {
     setGameStatus(GameStatus.inprogress);
     setCardsForPlayer(await cardsApi.getNewCard(deckId, 2));
+    setCountDealer(await setDealersHand());
   };
 
   const onSubmit = async () => {
     setCardsForPlayer(await cardsApi.getNewCard(deckId, 1));
   };
 
-  const setResultForPlayer = () => {
-    winner?.forEach((player) => {
-      player.name === email ? setPlayerWin(true) : setPlayerWin(false);
-    });
-    // do not changes into true
-  };
-
-  const onStopSubmit = async () => {
+  const onStopSubmit = () => {
     setGameStatus(GameStatus.finished);
 
-    setCountDealer(await setDealersHand());
-
     setWinner(findWinner());
 
-    setResultForPlayer();
+    setResultForPlayer(findWinner());
   };
 
   const createArrayOfAllPlayers = () => {
     const player = { name: email, id: uuidv4(), points: pointsPlayer };
-    const dealer = { name: "dealer", id: uuidv4(), points: countDealer }; ///!!!!!
+    const dealer = { name: "dealer", id: uuidv4(), points: countDealer };
+
     return [player, dealer, ...players];
   };
 
   const findWinner = () => {
     let winner: IPlayer[] = [];
 
     const players = createArrayOfAllPlayers();
```

Comment:
why you creating players in moment when you need to find winner?

Reply:
this is function is for creating array with all players(when they have already their final points) so due to which in another function i can find the winner based on their points

Reply:
But in every call createArrayOfAllPlayers dealer will have new id, it this may cause bugs in the future